### PR TITLE
Added automation for BZ1876927

### LINKED ISSUE #109: [closed loop] automate BZ 1876927
For more details and reproduction steps, see [BZ1876927](https://bugzilla.redhat.com/show_bug.cgi?id=1876927)

### PR DESCRIPTION
**Pull request by hkx303**
**Description**
Added automation for [BZ1876927](https://bugzilla.redhat.com/show_bug.cgi?id=1876927) #109 

**Test Result**
```
[hkx303@kuhuang tier2]$ nosetests tc_2066_validate_cluster_name_with_special_char.py 
2020-11-17 21:59:47 [INFO] ++++++++++++++++++++++++++++++
2020-11-17 21:59:47 [INFO] RHEL-196078:tc_2066_validate_cluster_name_with_special_char.py
2020-11-17 22:01:11 [INFO] Succeeded to get esxi host uuid: 2ccdb7e2-6bc3-493a-9431-9c1c3bf19490
2020-11-17 22:01:15 [INFO] Succeeded to get satellite host_id: bootp-73-130-244.rhts.eng.pek2.redhat.com:276
2020-11-17 22:01:25 [INFO] Succeeded to delete host: bootp-73-130-244.rhts.eng.pek2.redhat.com
2020-11-17 22:01:41 [INFO] Succeeded to unregister and clean system(ent-02-vm-06.lab.eng.nay.redhat.com)
2020-11-17 22:01:48 [INFO] Failed to recovery rhsm.conf, backup file is not found(ent-02-vm-06.lab.eng.nay.redhat.com)
2020-11-17 22:02:19 [INFO] Succeeded to register system ent-02-vm-06.lab.eng.nay.redhat.com to ent-02-vm-01.lab.eng.nay.redhat.com(satellite68-cdn-rhel7)
2020-11-17 22:02:29 [INFO] Succeeded to unregister and clean system(10.73.131.184)
2020-11-17 22:03:00 [INFO] Succeeded to register system 10.73.131.184 to ent-02-vm-01.lab.eng.nay.redhat.com(satellite68-cdn-rhel7)
2020-11-17 22:03:01 [INFO] Succeeded to disable all options in /etc/virt-who.conf
2020-11-17 22:03:04 [INFO] Succeeded to disable all options in /etc/sysconfig/virt-who
2020-11-17 22:03:06 [INFO] Succeeded to delete all files in /etc/virt-who.d/
2020-11-17 22:03:10 [INFO] Succeeded to create config file /etc/virt-who.d/virtwho-config.conf
2020-11-17 22:03:36 [INFO] Succeeded to get esxi host uuid: 2ccdb7e2-6bc3-493a-9431-9c1c3bf19490
2020-11-17 22:04:15 [INFO] >>>step1: change the vcenter cluster name to: virtwho/test
2020-11-17 22:04:51 [INFO] Succeeded to set the cluster name to virtwho/test
2020-11-17 22:04:51 [INFO] >>>step2: run virt-who service with the new cluster name
2020-11-17 22:05:09 [INFO] Start to run virt-who by service
2020-11-17 22:05:33 [INFO] pending_job: 0, is_429: no, loop_num: 0, loop_time: -1, send_num: 1, error_num: 0, thread_num: 1
2020-11-17 22:05:33 [INFO] virt-who is terminated by expected_send and expected_loop ready
2020-11-17 22:05:53 [INFO] virtwho thread number(1) is expected
2020-11-17 22:05:53 [INFO] virtwho error number(0) is expected
2020-11-17 22:05:53 [INFO] virtwho send number(1) is expected
2020-11-17 22:05:53 [INFO] Finished to validate all the expected options
2020-11-17 22:05:53 [INFO] >>>step3: check the hyperivsor fact by hammer command
2020-11-17 22:05:55 [INFO] Succeeded to get satellite host_id: bootp-73-130-244.rhts.eng.pek2.redhat.com:279
2020-11-17 22:05:58 [INFO] Succeed to get expected host info
2020-11-17 22:06:04 [INFO] Succeeded to search, expected msg(virtwho/test) is exist(1)
2020-11-17 22:06:04 [INFO] Succeeded to search, unexpected msg(virtwho-test) is not exist(0)
2020-11-17 22:06:04 [INFO] >>>step finally: change back the vcenter cluster name
2020-11-17 22:06:41 [INFO] Succeeded to set the cluster name to virtwho-test
2020-11-17 22:06:41 [INFO] Succeeded to run case, all steps passed

.
----------------------------------------------------------------------
Ran 1 test in 413.671s

OK

```
```
[hkx303@kuhuang tier2]$ pytest tc_2066_validate_cluster_name_with_special_char.py
==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-5.4.3, py-1.8.0, pluggy-0.13.1
rootdir: /home/hkx303/Documents/CI/virtwho-ci/tests/tier2
plugins: services-1.3.1, mock-1.10.4, forked-1.0.2, ibutsu-1.0.34, cov-2.7.1, xdist-1.34.0
collected 1 item                                                                             

tc_2066_validate_cluster_name_with_special_char.py .                                   [100%]

========================= 1 passed, 14 warnings in 365.79s (0:06:05) =========================
```

### CODE REVIEW

**Comment by hkx303** [comment]
A small question: Is this case should only run in Satellite or also should run in Stage?

File: tests/tier2/tc_2066_validate_cluster_name_with_special_char.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        results = dict()
+        config = self.get_hypervisor_config()
+        ssh_hypervisor = config['ssh_hypervisor']
+        register_config = self.get_register_config()
+        server = register_config['server']
+        ssh_user = register_config['ssh_user']
+        ssh_passwd = register_config['ssh_passwd']
+        ssh_register = {"host": server, "username": ssh_user, "password": ssh_passwd}
+        admin_user = register_config['username']
+        admin_passwd = register_config['password']
```
Comment (eko999) [inline tests/tier2/tc_2066_validate_cluster_name_with_special_char.py]: I guess we don't need to define the admin_user and admin_passwd for hammer  cli?

File: tests/tier2/tc_2066_validate_cluster_name_with_special_char.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        config_name = "virtwho-config"
+        config_file = "/etc/virt-who.d/{0}.conf".format(config_name)
```
Comment (eko999) [inline tests/tier2/tc_2066_validate_cluster_name_with_special_char.py:30]: should we enable the debug mode for console output?

File: tests/tier2/tc_2066_validate_cluster_name_with_special_char.py (hunk 1 above)
Comment (hkx303) [reply tests/tier2/tc_2066_validate_cluster_name_with_special_char.py]: Removed it.

File: tests/tier2/tc_2066_validate_cluster_name_with_special_char.py (hunk 2 above)
Comment (hkx303) [reply tests/tier2/tc_2066_validate_cluster_name_with_special_char.py:30]: Yes, we should. Thank you.

**Pull request merged**